src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -2240,13 +2240,8 @@ void CodeGen::genSSE42Intrinsic(GenTreeHWIntrinsic* node, insOpts instOptions)
         {
             assert(instOptions == INS_OPTS_NONE);
 
-#ifdef DEBUG
-            if (op2->isUsedFromReg() && (op2->GetRegNum() == targetReg) && (op1Reg != targetReg))
-            {
-                assert(genIsSameLocalVar(op1, op2));
-            }
-#endif
-
+            assert(!op2->isUsedFromReg() || (op2->GetRegNum() != targetReg) || (op1Reg == targetReg) ||
+                   genIsSameLocalVar(op1, op2));
```

Comment:
Something that's been nagging the back of my mind...

Do we know why LSRA is assigning a single local var to two registers simultaneously? If it's the same local and its the same instruction/use, then it really should've ensured that `op1Reg == op2Reg`, no?

That is, this seems like a general deficiency or issue with assignments leading to poorer codegen and higher register pressure.